tests/test_mousebind.py

```diff
@@ -0,0 +1,35 @@
+import pytest
+
+
+@pytest.mark.parametrize('method', ['-F', '--all'])
+def test_mouseunbind_all(hlwm, method, mouse):
+    hlwm.create_client()
+    hlwm.call('mousebind Button1 call quit')
+
+    unbind = hlwm.call(['mouseunbind', method])
+
+    assert unbind.stdout == ''
+    # TODO: assert hlwm.call('list_mousebind').stdout == ''
+    mouse.click('1')  # verify that binding got ungrabbed
+
+
+@pytest.mark.parametrize('button', [1, 2, 3, 4, 5])
+def test_trigger_mouse_binding_without_modifier(hlwm, mouse, button):
+    hlwm.call(f'mousebind Button{button} call close')
+    client_id, _ = hlwm.create_client()
+
+    mouse.click(str(button), client_id)
+
+    assert hlwm.get_attr('tags.0.client_count') == '0'
```

Comment:
This is very racy! `close` does not close the window but sends a request to the client to close its window. So it is very likely, that the next hlwm-command in the queue (the `get_attr`) is fully executed before the client fully handles the close-request and long before the notification about the closed window reaches the event queue in hlwm.

If you want to test the binding, you can create a custom attribute (`new_attr string my_test`) and set this attribute in the mouse binding (`set_attr my_test something`), in order to assert its value

Reply:
Thanks for noticing! The `set_attr` approach is a good idea, I'll change it.